### PR TITLE
Add more Go samples for BigQuery:

### PR DESCRIPTION
**Pull request by shollyman**
bigquery_extract_table
bigquery_extract_table_json

Additionally, make error reporting in snippet_test.go more informative.

### CODE REVIEW

**Comment by shollyman** [comment]
test failure appears to be infrastructure related (lack of free test projects available for lease) rather than a code failure

File: bigquery/snippets/snippet.go [hunk 1]
```diff
@@ -379,6 +379,101 @@ func exportToGCS(client *bigquery.Client, datasetID, tableID, gcsURI string) err
 	return nil
 }
 
+func exportSampleTableAsCSV(client *bigquery.Client, gcsURI string) error {
```
Comment (tbpg) [inline bigquery/snippets/snippet.go]: Can we test these new functions?

Comment (shollyman) [reply bigquery/snippets/snippet.go]: They're invoked in snippet_test.go.  What kind of test are you looking for?  Probe the bucket for the file(s)?

Comment (tbpg) [reply bigquery/snippets/snippet.go]: Whoops! You're absolutely right - I was looking at a subset of changes. LGTM.

**Pull request merged**